.ycm_extra_conf.py

```diff
@@ -36,7 +36,7 @@
 # CHANGE THIS LIST OF FLAGS. YES, THIS IS THE DROID YOU HAVE BEEN LOOKING FOR.
 flags = [
     '-DNDEBUG',
-    '-DWITH_TBB',
+    '-DNMC_HAVE_TBB',
```

Comment:
The explanation goes like this:
A user says I want Option X (say cuda or openMP) so in CMake they say NMC_WITH_CUDA or NMC_WITH_OPENMP or something of that kind. (ignoring the fact that you have a threading model var for that). It might be that OpenMP is not found (OSX) but it isn't fatal, so you don't mind. Elsewhere in the code there will be checks for if(openmp) etc, which should of course test negative, so a common pattern is to have user facing options WITH_XXX and the actual vars that code checks to be HAVE_XXX so that if(WITH_openmp) passes add_definition(HAVE_openmp) is added. 
User facing options use WITH, actual ifdefs in the code use HAVE to perform the checks (and if cmake needs it, it can also have HAVE_option vars set for various checks).
The commits themselves have quite long log messages with some explanations too.

If you don't like using WITH and HAVE, then a search/replace can change all the _HAVE_back to _WITH_ in a jiffy - (but I'd like to convince you to keep it)